src/Approximations/template_directions.jl

```diff
@@ -330,3 +337,139 @@ end
 
 end # @eval
 end # if
+
+# ==================================================
+# Spherical directions
+# ==================================================
+
+# returns `true` if and only if there is a vector in the list of vectors S
+# such that it is approximately d
+function _isapprox_included(d::Vector{N}, S::Vector{Vector{N}}) where {N}
+    result = false
+    for si in S
+        if isapproxzero(norm(d-si))
+            result = true
+            break
+        end
+    end
+    return result
+end
+
+"""
+    SphericalDirections{N} <: AbstractDirections{N}
+
+Spherical directions representation.
+
+### Fields
+
+- `Nθ`    -- length of the partition of the azimuthal angle
+- `Nφ`    -- length of the partition of the polar angle
+- `stack` -- list of computed directions
+
+### Notes
+
+The `SphericalDirections` constructor provides a sample of the unit sphere
+in ``\\mathbb{R}^3``, which is parameterized by the azimuthal and polar angles
+``θ ∈ Dθ := [-π, π]`` and ``φ ∈ Dφ := [0, 2π]`` respectively, see the wikipedia
+entry [Spherical coordinate system](https://en.wikipedia.org/wiki/Spherical_coordinate_system).
+The domains ``Dθ`` and ``Dφ`` are discretized in ``Nθ`` and ``Nφ`` respectively.
+Then the Cartesian componentes of each direction are obtained with
+
+```math
+[sin(θᵢ)*cos(φᵢ), sin(θᵢ)*sin(φᵢ), cos(θᵢ)]
+```
+Note that by discretization, the polar point may be computed more than once, in
+general. Removing duplicates takes time; you can choose to remove duplicates or
+not by setting the keyword argument `remove_duplicates` in the constructor
+(default `true`).
+
+### Examples
+
+A `SphericalDirections` can be built in different ways. If you pass only one integer,
+it is used to discretize both ``θ`` and ``φ``:
+
+```jldoctest spherical_directions
+julia> using LazySets.Approximations: SphericalDirections
+
+julia> sd = SphericalDirections(3)
+SphericalDirections(3, 3, Array{Float64,1}[[-1.22465e-16, -0.0, -1.0], [0.0, 0.0, 1.0]])
+
+julia> sd.Nθ, sd.Nφ 
+(3, 3)
+```
+
+Repeated directions are removed by default. Use the keyword argument `remove_duplicates`
+to control this behavior:
+
+```jldoctest spherical_directions
+julia> sd_without_duplicates = SphericalDirections(3, remove_duplicates=true);
+
+julia> length(sd_without_duplicates)
+2
+
+julia> sd_with_duplicates = SphericalDirections(3, remove_duplicates=false);
+
+julia> length(sd_with_duplicates)
+9
+```
+
+Pass two integers to control the discretization in ``θ`` and in ``φ`` separately:
+```jldoctest spherical_directions
+julia> sd_4_5 = SphericalDirections(4, 5);
+
+julia> length(sd_4_5)
+5
+
+julia> sd_4_8 = SphericalDirections(4, 8);
+
+julia> length(sd_4_8)
+15
+```
+"""
+struct SphericalDirections{N} <: AbstractDirections{N}
+    Nθ::Int
+    Nφ::Int
+    stack::Vector{Vector{N}} # stores the spherical directions
+
+    function SphericalDirections(Nθ::Int, Nφ::Int; N::Type{<:AbstractFloat}=Float64, remove_duplicates=true)
```

Comment:
Either you restrict the type of `N` in the struct or you should omit the restriction here.

Reply:
By omit the restriction you mean to use `N::Type{<:Number}=Float64` here?

Reply:
I mean to have the same restriction as in the struct. There are no restrictions, so there should be no restrictions here as well.
Also, it is better to use `function SphericalDirections{N}(...) where {N}` instead of adding the type as a normal argument.

Reply:
Ok, got it. I've used `AbstractFloat` consistently now. I noted that using exact arithmetic and the trigonometric functions don't play well, for instance `Rational{Int}` easily overflows already `Nθ=5`. Since we don't test with `Rational{BigInt}` i would just leave it like that.. If usage arises we can remove the annotation..